charts/drax/configuration/grafana/alerting/contactpoints.yaml

```
@@ -1,15 +1,23 @@
+{{- if $.Values.global.alerting.enabled -}}
+{{- if eq $.Values.global.alerting.channel "" -}}
+  {{- fail "A slack channel needs to be provided when alerting is enabled (global.alerting.channel). Please set a valid Slack channel name or Slack ID." }}
+{{- end -}}
+{{- end -}}
+
 apiVersion: 1
 contactPoints:
   - orgId: 1
     name: accelleran-drax
     receivers:
+      {{- if and ($.Values.global.alerting.enabled) (ne $.Values.global.alerting.channel "") }}
```

```suggestion
      {{- if and ((($.Values.global).alerting).enabled) (ne (($.Values.global).alerting).channel "") }}
```

---

Is there a specific reason why we need to use extra parenthesis?

---

It will evaluate to nil instead of resulting in a deref nil panic potentially. It's only for edge cases though which I don't think we will ever run into, but I like it to be a bit safer where possible.

---

I just notice as well that instead of checking that it's not equal to the empty string it's probably better to check emptyiness:

```suggestion
      {{- if and ((($.Values.global).alerting).enabled) (empty (($.Values.global).alerting).channel) }}
```

---

My last comment was not tackled.